### PR TITLE
chore: Base Distributor and related simple redeem/supply test cases 

### PR DESCRIPTION
**Pull request by charleenfei**

### CODE REVIEW

File: src/core/test/system/lender/base/redeem_single_tranche.t.sol [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        uint redeemAmount = 5 ether;
+        supply(investorBalance, supplyAmount);
+        // first redeem
+        juniorInvestor.doRedeem(redeemAmount);
+        // second redeem
+        juniorInvestor.doRedeem(redeemAmount);
+        assertPostCondition(investorBalance);
+    }
+
+    function assertPostCondition(uint investorBalance) public {
```
Comment (manuelpolzhofer) [inline src/core/test/system/lender/base/redeem_single_tranche.t.sol]:
```suggestion
    function assertPostCondition(uint investorBalance, uint investorTokenAmount) public {
```

File: src/core/test/system/lender/base/redeem_single_tranche.t.sol [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    function testPartialRedeem() public {
+        uint investorBalance = 100 ether;
+        uint supplyAmount = 10 ether;
+        uint redeemAmount = 5 ether;
+        supply(investorBalance, supplyAmount);
+        // first redeem
+        juniorInvestor.doRedeem(redeemAmount);
+        // second redeem
```
Comment (manuelpolzhofer) [inline src/core/test/system/lender/base/redeem_single_tranche.t.sol]: lets check here the post condition

File: src/core/test/system/lender/base/redeem_single_tranche.t.sol [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    function assertPostCondition(uint investorBalance) public {
+        // assert: no more tokens left for junior investor
+        assertEq(juniorERC20.balanceOf(juniorInvestor_), 0);
+        // assert: back to original balance
+        assertEq(currency.balanceOf(juniorInvestor_), investorBalance);
+        // assert: shelf is balanced, excess has either been transferred to tranches or needed money transferred from distributor -> shelf
+        assertEq(shelf.balance() - currency.balanceOf(address(shelf)), 0);
+    }
+
+    function testFailNotEnoughMoney() public {
```
Comment (manuelpolzhofer) [inline src/core/test/system/lender/base/redeem_single_tranche.t.sol]:
```suggestion
    function testFailNotEnougToken() public {
```

File: src/core/test/system/lender/base/supply_single_tranche.t.sol [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        currency.mint(juniorInvestor_, investorBalance);
+
+        juniorInvestor.doSupply(supplyAmount);
+        assertPostCondition(supplyAmount);
+        assertEq(currency.balanceOf(address(junior)), supplyAmount);
+    }
+
+    function assertPostCondition(uint supplyAmount) public {
+        // assert: shelf is balanced, excess has either been transferred to tranches or needed money transferred from distributor -> shelf
+        assertEq(shelf.balance() - currency.balanceOf(address(shelf)), 0);
```
Comment (manuelpolzhofer) [inline src/core/test/system/lender/base/supply_single_tranche.t.sol]: Is it always guaranteed?  Lets say the admin approves higher principals than volume in the tranches

Comment (charleenfei) [reply src/core/test/system/lender/base/supply_single_tranche.t.sol]: just for these post conditions. I will do more complex scenarios in another PR

**Pull request merged**